src/filesystem/File.js

```diff
@@ -122,14 +129,15 @@ define(function (require, exports, module) {
             // Always store the hash
             this._hash = stat._hash;
             this._encoding = encoding;
+            this._preserveBOM = preserveBOM;
 
             // Only cache data for watched files
             if (watched) {
                 this._stat = stat;
                 this._contents = data;
             }
 
-            callback(err, data, encoding, stat);
+            callback(err, data, encoding, this._preserveBOM, stat);
```

Comment:
Remove the preserveBOM marker from callback. It should be consumed internally and kept in File prototype only.